libs/optimizer/src/lib/optimizer.ts

```diff
@@ -46,7 +46,7 @@ export interface ScoringResult {
   lengthKm: number;
   // Multiplier for computing score. score = lengthKm * multiplier
   multiplier: number;
-  circuit?: CircuitType;
+  circuit: CircuitType;
```

Comment:
Do I remember correctly that you test for this being undefined at some point in this PR.
You can remove that test

Reply:
In fact the undefined case was not tested.(we use toMatchObject which is some kind of lenient check).
 I changed the test and added the value (CircuitType.OpenDistance) expected when scoring an empty track